test/libyul/Common.cpp

```diff
@@ -41,6 +41,14 @@ using namespace langutil;
 using namespace yul;
 using namespace dev::solidity;
 
+namespace
+{
+shared_ptr<Dialect> defaultDialect(bool _yul)
```

Comment:
Not strictly related to this PR, but could this be a reference instead of a ``shared_ptr``? I.e. is there a reason for ``yul::Dialect::yul()`` and ``yul::EVMDialect::structAssemblyForEVM()`` not to return references instead?
Especially since dialects are supposed to be immutable anyways. But yeah, not sure and probably not in scope for this PR anyways.

Reply:
EVMDialect is actually mutable... ;)

Reply:
Ah, ok :-).